src/game.cpp

```diff
@@ -10872,24 +10872,45 @@ void game::chat()
     }
 
     uimenu nmenu;
-    nmenu.text = std::string( _("Who do you want to talk to?") );
-
+    nmenu.text = std::string( _( "Who do you want to talk to or yell at?" ) );
+    
     int i = 0;
+
     for( auto &elem : available ) {
         nmenu.addentry( i++, true, MENU_AUTOASSIGN, ( elem )->name );
     }
 
     nmenu.return_invalid = true;
-    nmenu.addentry( i++, true, 'a', _( "Yell" ) );
-    nmenu.addentry( i++, true, 'q', _( "Cancel" ) );
+
+    int yell, yell_sentence, cancel;
+
+    nmenu.addentry( yell = i++, true, 'a', _( "Yell" ) );
```

Comment:
@AlecWhite

You can create `enum choices { yell, yell_sentence, cancel }` and use -n for selecting the 1st NPC, -n + 1 for selecting the 2nd one, etc., and -1 for selecting the nth one. It will preserve the menu ordering you want. But, in my opinion, it would over-complicate the code instead of making it simpler.

@codemime

> You can use negative values to avoid overlapping with indexes of NPCs:

Are negative numbers sorted after non-negative by the game's engine? If no, what you propose won't work.